keps/1833-metrics-for-local-queue/README.md

```diff
@@ -0,0 +1,165 @@
+currently provides the ability to enable collection of metrics for cluster queues. This API will be extended to include options for enabling metrics 
+collection for local queues.
+
+The `ControllerMetrics` that contain the option to configure metrics, will be extended as follows:
+
+```go
+type ControllerManager struct {
+	...
+
+	// Metrics contains the controller metrics configuration
+	// +optional
+	Metrics ControllerMetrics `json:"metrics,omitempty"`
+	...
+}
+
+// ControllerMetrics defines the metrics configs.
+type ControllerMetrics struct {
+	...
+	
+	// EnableLocalQueueResources, if true the local queue resource usage and quotas
+	// metrics will be reported.
+	// +optional
+	EnableLocalQueueResources bool `json:"enableLocalQueueResources,omitempty"`
+	
+	// LocalQueueMetricOptions specifies the configuration options for local queue metrics.
+	LocalQueueMetricOptions *metricsOptions `json:"localQueueMetricOptions,omitempty"`
+}
+
+// metricsOptions defines the configuration options for local queue metrics.
+// If left empty, then metrics will expose for all local queues across namespaces.
+type metricsOptions struct {
+	// NamespaceSelector can be used to select namespaces in which the local queues should
+	// report metrics.
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+	
+	// QueueSelector can be used to choose the local queues that need metrics to be collected. 
+	QueueSelector *metav1.LabelSelector `json:"queueSelector,omitempty"`
+}
+```
+
+To reduce cardinality, and enable selection of metrics for local queues, the following
+knobs will be available:
+
+1. If `EnableLocalQueueResources` is false, then metrics will not be exposed.  
+1. If `EnableLocalQueueResources` is true, and `LocalQueueMetricOptions` is **not** provided - metrics will be exposed for all local queues.
+1. If `EnableLocalQueueResources` is true and `LocalQueueMetricOptions`  is provided - metrics will be collected for local queues that match specified label selectors.
+
+### List of metrics for Local Queues:
+
+In the first iteration, following are the list of metrics that would contain information on Local Queue statuses:
+
+| Metrics Name                   | Prometheus Type | Description                                                 | 
+|--------------------------------|-----------------|-------------------------------------------------------------|
+| local_queue_pending_workloads  | Gauge           | The number of pending workloads                             |
+| local_queue_reserved_workloads | Counter         | Total number of workloads in the LocalQueue reserving quota |
+| local_queue_admitted_workloads | Counter         | Total number of admitted workloads                          |
+| local_queue_resource_usage     | Gauge           | Total quantity of used quota per resource for a Local Queue |
+
```

Comment:
Could we also add metrics such as `local_queue_evicted_workloads`, `local_queue_admission_wait_time_seconds`,  `local_queue_reserved_wait_time_seconds` similarly to https://kueue.sigs.k8s.io/docs/reference/metrics/#clusterqueue-status

Reply:
@PBundyra I was hoping to add this in the next iteration, to make the initial set minimal and just introduce this feature. But probably, worth adding this to the KEP and just separating it out during implementation to make it easier for reviews.

Reply:
Addressed in https://github.com/kubernetes-sigs/kueue/pull/2516/commits/e86c944647715169fc388477e72c75051694d480

Reply:
Thank you, can we also add `local_queue_wait_time_seconds`?

Reply:
Could that not be covered by admission and reserved wait time parameters?

Reply:
It would, but I imagine it would be hard to sum up those times to get aggregated results

Reply:
Updated.